File: 3493-maximum-number-of-operations-to-move-ones-to-the-end/3493-maximum-number-of-operations-to-move-ones-to-the-end.java
class Solution {
    public int maxOperations(String s) {
        int oneCount =0;
        int i=0;
        int ans=0;
        int n=s.length();
        while(i<n){
            if(s.charAt(i)=='0'){
                while(i+1<n && s.charAt(i+1)=='0'){
                    i++;
                }
                ans+=oneCount;
            }
            else{
                oneCount++;
            }
            i++;
        }
        return ans;
    }
}

File: 3493-maximum-number-of-operations-to-move-ones-to-the-end/3493-maximum-number-of-operations-to-move-ones-to-the-end.py
class Solution:
    def maxOperations(self, s: str) -> int:
        ans=0
        one_count=0
        i=0
        n=len(s)
        while i< n:
            ch=s[i]
            if ch=='0':
                while i+1<n and s[i+1]=='0':
                    i+=1
                ans+=one_count
            else:
                one_count+=1
            
            i+=1
        return ans

        